packages/terra-paginator/src/_PaginatorButton.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { disableFocusActiveStyles, disableFocusOnBlur, enableFocusActiveStyles } from './_paginationUtils';
+import styles from './Paginator.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Sets aria-current attribute.
+   */
+  ariaCurrent: PropTypes.oneOfType([
+    PropTypes.bool,
+    PropTypes.string,
+  ]),
+  /**
+   * Sets aria-disabled attribute.
+   */
+  ariaDisabled: PropTypes.bool,
+  /**
+   * Sets aria-label attribute.
+   */
+  ariaLabel: PropTypes.string,
+  /**
+   * Child elements passed in to buttons.
+   */
+  children: PropTypes.node,
+  /**
+   * Passed in CSS className.
+   */
+  className: PropTypes.string,
+  /**
+   * Callback function triggered when clicked.
+   */
+  onClick: PropTypes.func,
+};
+
+class PaginatorButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleFocus = this.handleFocus.bind(this);
+    this.handleMouseDown = this.handleMouseDown.bind(this);
+
+    this.shouldShowFocus = true;
```

Comment:
You could utilize `useRef` for this and make this a functional component.

Reply:
Migrated the component over to the functional variety: [30998be](https://github.com/cerner/terra-core/pull/2686/commits/30998bee8fb4b9c77312b0b144243e42d0dc274f). The effect functions just using a local variable for `shouldShowFocus` since it is just being used (in this case) as a dumb flag, but if there are other issues with it I have missed, let me know!